ads/widespace.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {writeScript, validateDataExists, checkData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function widespace(global, data) {
+  checkData(data, ['sid']);
+  validateDataExists(data, ['sid']);
+
+  const url = 'https://engine.widespace.com/map/engine/dynamic?isamp=1&sid=' + encodeURIComponent(data.sid);
+
+  writeScript(global, url);
```

Comment:
Yes I know, we have two kinds of implementations. 
This one that we use in iframe that are more simpel but use document.write. And one that is async but is larger and create its own sandboxed iframe and expose an api. 

On AMP we think the the lighter implementation with document.write will work better as we are in your iframe anyway and we lose the benefits with our api etc. Also to have your iframe over our sandboxed iframe feels little wrong. So if we can use this implementation I think it is better.